### DISCUSSION TITLE
Get email analytics detail with the API

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [buttondown/discussions#25] [Q&A] [answered]
I can see get analytics using the API.

```shell
recipients           : 822
deliveries           : 822
opens                : 399
clicks               : 19
temporary_failures   : 7
permanent_failures   : 2
unsubscriptions      : 0
complaints           : 0
survey_responses     : 0
webmentions          : 0
page_views__lifetime : 0
page_views__30       : 0
page_views__7        : 0
subscriptions        : 0
paid_subscriptions   : 0
replies              : 0
comments             : 0
```

This shows me a number. However, I learned the other day I can get details in the web UI.

![image](https://github.com/buttondown/discussions/assets/12089920/268fc8ac-044d-4808-ae17-41d21fb20ee1)

Can I get this same information through the API? I am most concerned about failure details. I'm trying to build an automated reporting system.

**Comment by jmduke** [comment, marked as answer]
@jdhitsolutions You sure can! You'll want to call:

```
/v1/emails/<id>/events?event_type=bounced
```

(A fun fact — almost everything in the UI is powered by the external API. When in doubt, pop open the inspector tab and have a poke around!)

**Reply by jdhitsolutions** [reply]
Thanks. I would never have made that connection by reading the documentation.

**Reply by jmduke** [reply]
The API docs are in a bad state right now (they're accurate, but very hard to navigate). They'll be better in April!

**Comment by jdhitsolutions** [comment]
How would I have discovered the `bounced` event type? I don't see it listed in the event and webhook list. The reason I ask is that this shows me temporary failure details, but it doesn't seem to give me permanent failure details. I'm assuming that is a different event type.

**Reply by jdhitsolutions** [reply]
I went back to the web browser and played around more with the Inspector. Not something I have much experience with. I discovered the event type I'm looking for is `rejected`.  The metadata property is empty, which I'm assuming is why you show `unknown` in the web UI.

**Reply by jmduke** [reply]
All great points. Will make the following changes:

1. Add a blurb to `events-and-webhooks` pointing to "email events" (which are, sadly, a slightly different concept)
2. Add a tutorial/quickstart guide for programmatically retrieving analytics data
3. Add a page enumerating various email event types.

**Reply by jdhitsolutions** [reply]
I apologize for the nuisance. For all I know, I may be the only one using the API for this task. I downloaded the schema YAML, but that didn't help.

**Reply by jmduke** [reply]
This is great feedback — not a nuisance at all! Keep it coming :)
